src/LibraryManager/CacheService/CacheFileMetadata.cs

```diff
@@ -28,5 +30,23 @@ public CacheFileMetadata(string source, string cacheFile)
         /// Source for the cache's contents  
         /// </summary>
         public string Source { get; private set; }
+
+        /// <inheritdoc />
+        public bool Equals(CacheFileMetadata other)
+        {
+            return DestinationPath == other.DestinationPath && Source == other.Source;
```

Comment:
Yes, since we could be running on case-sensitive filesystems.  Technically things could get messy if a library is installed on a case-instensitive filesystem and contains differently-cased files with different contents, but I don't think we have a good way to handle that (and if we did it's probably in a different code path, like where we validate the sanity of the libman.json manifest).